src/games/strategy/net/ClientLoginHelper.java

```diff
@@ -24,15 +25,18 @@ public boolean login() {
       // write the object output streams magic number
       out.flush();
       final ObjectInputStream in = new ObjectInputStream(streams.getBufferedIn());
-      @SuppressWarnings("rawtypes")
-      final Map challenge = (Map) in.readObject();
+      Object challengeObject = in.readObject();
+      Map<String, String> challenge = null;
+      if(challengeObject instanceof Map){
+        challenge = (Map<String, String>) challengeObject;
+      }
```

Comment:
Are these check really worth it? Would it be better to just has it to Map<String, String>?